types/pdfmake/interfaces.d.ts

```diff
@@ -0,0 +1,473 @@
+    tocStyle?: string | string[] | Style;
+    tocNumberStyle?: string | string[] | Style;
+    tocMargin?: Margins;
+}
+
+export interface ContentPageReference extends ContentBase {
+    pageReference: string;
+}
+
+export interface ContentTextReference extends ContentBase {
+    textReference: string;
+}
+
+export interface ContentToc extends ContentBase {
+    toc: TableOfContent;
+}
+
+export interface ContentQr extends ContentBase {
+    qr: string;
+    foreground?: string;
+    fit?: number;
+    version?: number;
+    eccLevel?: 'L' | 'M' | 'Q' | 'H';
+    mode?: 'numeric' | 'alphanumeric' | 'octet';
+    mask?: number;
+}
+
+export interface ContentBase extends Style {
+    style?: string | string[] | Style;
+    absolutePosition?: { x: number; y: number };
+    relativePosition?: { x: number; y: number };
+    pageBreak?: 'before' | 'after';
+    pageOrientation?: PageOrientation;
+    headlineLevel?: number;
+}
+
+export interface ContentLink {
+    link?: string;
+    linkToPage?: number;
+    linkToDestination?: string;
+}
+
+export interface TableOfContent {
+    title?: Content;
+    textMargin?: Margins;
+    textStyle?: string | string[] | Style;
+    numberStyle?: string | string[] | Style;
+    id?: string;
+}
+
+export type Column = Content & {
+    width?: Size;
+};
+
+export type OrderedListType = 'lower-alpha' | 'upper-alpha' | 'lower-roman' | 'upper-roman' | 'none';
+export type OrderedListElement = Content & {
+    counter?: number;
+    listType?: OrderedListType;
+};
+
+export type UnorderedListType = 'square' | 'circle' | 'none';
+export type UnorderedListElement = Content & {
+    listType?: UnorderedListType;
+};
+
+export type CanvasElement = CanvasRect | CanvasPolyline | CanvasLine | CanvasEllipse;
+
+export interface CanvasRect extends CanvasLineElement, CanvasFilledElement {
+    type: 'rect';
+    x: number;
+    y: number;
+    w: number;
+    h: number;
+    r?: number;
+}
+
+export interface CanvasPolyline extends CanvasLineElement, CanvasFilledElement {
+    type: 'polyline';
+    points: Array<{ x: number; y: number }>;
+    closePath?: boolean;
+    lineCap?: 'round' | 'square';
+}
+
+export interface CanvasLine extends CanvasLineElement {
+    type: 'line';
+    x1: number;
+    y1: number;
+    x2: number;
+    y2: number;
+    lineCap?: 'round' | 'square';
+}
+
+export interface CanvasEllipse extends CanvasLineElement, CanvasFilledElement {
+    type: 'ellipse';
+    x: number;
+    y: number;
+    r1: number;
+    r2?: number;
+}
+
+export interface CanvasFilledElement {
+    color?: string;
+    fillOpacity?: number;
+    linearGradient?: string[];
+}
+
+export interface CanvasLineElement {
+    lineWidth?: number;
+    lineColor?: string;
+    dash?: {
+        length: number;
+        space?: number;
+    };
+}
+
+export interface StyleDictionnary {
+    [name: string]: Style;
+}
+
+export type PDFVersion = '1.3' | '1.4' | '1.5' | '1.6' | '1.7' | '1.7ext3';
+
+export interface Watermark {
```

Comment:
Please add back type annotation comments.

Reply:
I didn't pay attention while merging, will do.

Reply:
This looks good.